shared/js/background/atb.es6.js

```diff
@@ -146,19 +146,27 @@ var ATB = (() => {
             // - the user wasn't already looking at the app install page
             // - the user hasn't seen the page before
             settings.ready().then( () => {
-                chrome.tabs.query({currentWindow: true, active: true}, function(tabs) { 
+                chrome.tabs.query({currentWindow: true, active: true}, function(tabs) {
                     const domain = (tabs && tabs[0]) ? tabs[0].url : ''
-                    const regExpPostInstall = new RegExp('duckduckgo\.com\/app')
-                    if ((!settings.getSetting('hasSeenPostInstall')) && (!domain.match(regExpPostInstall))) {
-                        settings.updateSetting('hasSeenPostInstall', true)
-                        chrome.tabs.create({
-                            url: 'https://duckduckgo.com/app?post=1'
-                        })
+                    if (this._canShowPostInstall(domain)) {
+                            settings.updateSetting('hasSeenPostInstall', true)
+                            chrome.tabs.create({
+                                url: 'https://duckduckgo.com/app?post=1'
+                            })
                     }
                 })
             })
         },
 
+        _canShowPostInstall: (domain) => {
+            const regExpPostInstall = /duckduckgo\.com\/app/
+            const regExpSoftwarePage = /duckduckgo\.com\/software/
+
+            return !settings.getSetting('hasSeenPostInstall')
+                && !domain.match(regExpPostInstall)
```

Comment:
Could you also check whether settings and domain are defined, please. If not, return early